v3/doc/map.md

```diff
@@ -0,0 +1,79 @@
+
+This problem is addressed by making a temporary `LeafletMap` object that is the final size of the Map and then calling `fitBounds` on that temporary `LeafletMap`. Then we get the center and zoom of the temporary `LeafletMap` and apply those values immediately to the actual `LeafletMap` via `LeafletMapState.adjustMapView`. These center and zoom values are basically preserved as the map animates into place.
+
+There is a wrinkle with this approach when the plugin API creates a map with a legend. This legend takes up more the height of the tile, so the height of the map is smaller. It isn't easy to know this legend height when creating the temporary `LeafletMap`. This issue seems minor and plugins can work around it by setting the center and zoom of the map themselves.
+
+Originally this problem was solved by trying to wait for the animation to finish before calling `fitBounds`. It is very difficult to make that work correctly. We know when the CSS transition ends, but the actual size of the Leaflet container isn't updated until a resize observer sees the new tile size and updates the data configuration layout, which triggers the CodapMap to re-render and use the new size. On top of this, while the map tile is animating into place `LeafletMapState.adjustMapView({invalidateSize: true})` is being called each time the `CodapMap` re-renders during the animation. This is done so Leaflet re-renders in its new space. However each of these calls to `invalidateSize` causes Leaflet to fire `move` events which slightly adjust the center of the map. This eventually causes the center and zoom properties of `LeafletMapState` to be sync'd to the `MapContentModel`. Because of the debouncing described above this syncing might not actually take effect until after the animation completes but if there are are slow downs in the browser the syncing might happen during the animation. The result of all of this is that when the animation is done, sometimes the zoom of the `MapContentModel` would remain the default `-1` value and sometimes it would be set to `0`. That makes it hard to know if we should rescale the map at the end of the animation or not. It is possible to work around these nuances, but this is more complicated than the temporary leaflet approach. Additionally the temporary leaflet approach should result in a nicer looking animation.
+
+## Undo Redo
+The `LeafletMapState` object stores a undoStringKey and redoStringKey. Whenever the component calls the actions on `LeafletMapState` it will set these keys if the change should be undo-able. Then when the reaction in the map content model observes a change in `LeafletMapState` it uses these keys to record the change in the content model as undo able.
```

Comment:
```suggestion
The `LeafletMapState` object stores an undoStringKey and a redoStringKey. Whenever the component calls the actions on `LeafletMapState` it will set these keys if the change should be undo-able. Then when the reaction in the map content model observes a change in `LeafletMapState` it uses these keys to record the change in the content model as undo-able.
```